lib/common/xml.c

```diff
@@ -3895,7 +3923,8 @@ crm_xml_dump(xmlNode * data, int options, char **buffer, int *offset, int *max,
             dump_xml_element(data, options, buffer, offset, max, depth);
             break;
         case XML_TEXT_NODE:
-            /* Ignore */
+            /* Handle the text node*/
+            dump_xml_text(data, options, buffer, offset, max, depth);
```

Comment:
Most Pacemaker XML doesn't use text inside tags, so I'm not sure it's absolutely necessary, but in general we do try to preserve API behavior. Typically in a case like this, we'd add a new function as suggested, but perhaps crm_xml_dump_full() that takes one additional boolean parameter dump_text. Then the old function becomes a wrapper for the new one with dump_text=FALSE.

The "_full" suffix is a common convention in pacemaker code for situations like this. It's not the prettiest arrangement but it keeps backward compatibility without duplicating code.